### PR TITLE
Finalize NRSE

### PR DESCRIPTION
**Pull request by mgovers**
* [x] prepare_matrix_and_rhs:
  * [x] BUG: (L250) also clear fill-in
  * [x] (L328) no prefactorize is needed.
* [x] The NRSEGainBlock (L64-L67):
  * [x] qt should be (hermitian) transposed
  * [x] qt should be directly obtained from hermitian transpose instead of per element
* [x] Run_state_estimation:
  * [x] (L163) run solve directly using `prefactorize_and_solve`
* [x] fix dense grouped index vector not initialized by default
* [x] sensor-update case for NRSE
* [x] distribution-case for NRSE and iterative linear
* [x] single-line-load case

Relates to #369

### CODE REVIEW

**Comment by TonyXiang8787** [comment]
After examination I have concluded that the case `sensor-update` has indeed another local minimum for NRSE. Thus a new test case created.

**Comment by mgovers** [comment]
> After examination I have concluded that the case `sensor-update` has indeed another local minimum for NRSE. Thus a new test case created.

ok, clear. i'm working on the block matrix thingy. should i do that in this PR or a separate one?

**Comment by TonyXiang8787** [comment]
> > After examination I have concluded that the case `sensor-update` has indeed another local minimum for NRSE. Thus a new test case created.
> 
> ok, clear. i'm working on the block matrix thingy. should i do that in this PR or a separate one?

Just push to here. We try to finish everything.

**Comment by mgovers** [comment]
> > > After examination I have concluded that the case `sensor-update` has indeed another local minimum for NRSE. Thus a new test case created.
> > 
> > 
> > ok, clear. i'm working on the block matrix thingy. should i do that in this PR or a separate one?
> 
> Just push to here. We try to finish everything.

i would like to do the version bump + remove experimental flag in a separate PR, just to really make sure that that version bump is extremely minimal and that we don't end up finding something **again** before pushing

**Review by TonyXiang8787 — DISMISSED** [review]

File: tests/data/state_estimation/distribution-case/params.json [hunk 1]
```diff
@@ -1,16 +1,8 @@
 {
-  "calculation_method": ["iterative_linear", "newton_raphson"],
+  "calculation_method": "iterative_linear",
```
Comment (mgovers) [inline tests/data/state_estimation/distribution-case/params.json:2]: you disabled `newton_raphson`. that was not intended, right?

**Review by TonyXiang8787 — APPROVED** [review]

**Comment by mgovers** [comment]
I see no more weird/unexpected things. CI appears to go OK both locally and remotely. Hereby enabling automerge

**Pull request merged**